api/src/paths/xlsx/transform.ts

```diff
@@ -0,0 +1,260 @@
+      throw error;
+    }
+  };
+}
+
+function transformXLSX(): RequestHandler {
+  return async (req, res, next) => {
+    defaultLog.debug({ label: 'transformXLSX', message: 'xlsx transform' });
+
+    try {
+      const xlsxCsv: XLSXCSV = req['xlsx'];
+
+      const transformationSchemaParser: TransformationSchemaParser = req['transformationSchemaParser'];
+
+      const xlsxTransformation = new XLSXTransformation(transformationSchemaParser, xlsxCsv);
+
+      const transformedData = xlsxTransformation.transform();
+
+      const worksheets = xlsxTransformation.dataToSheet(transformedData);
+
+      const fileBuffers = Object.entries(worksheets).map(([fileName, worksheet]) => {
+        return {
+          name: fileName,
+          buffer: xlsxCsv.worksheetToBuffer(worksheet)
+        };
+      });
+
+      req['fileBuffers'] = fileBuffers;
+
+      next();
+    } catch (error) {
+      defaultLog.debug({ label: 'transformXLSX', message: 'error', error });
+      throw error;
+    }
+  };
+}
+
+export function persistTransformationResults(): RequestHandler {
+  return async (req, res, next) => {
+    try {
+      defaultLog.debug({ label: 'persistTransformationResults', message: 'xlsx transform' });
+
+      const fileBuffers: { name: string; buffer: Buffer }[] = req['fileBuffers'];
+
+      // Build the archive zip file
+      const dwcArchiveZip = new AdmZip();
+      fileBuffers.forEach((file) => dwcArchiveZip.addFile(`${file.name}.csv`, file.buffer));
+
+      // Build output s3Key based on the original input s3Key
+      const s3Key: string = req['s3Key'];
+      const outputS3KeyPrefix = s3Key.split('/').slice(0, -1).join('/'); // Remove the filename from original s3Key
+
+      const xlsxCsv: XLSXCSV = req['xlsx'];
+      const outputFileName = `${xlsxCsv.rawFile.name}.zip`;
+
+      const outputS3Key = `${outputS3KeyPrefix}/${outputFileName}`;
+
+      // Upload transformed archive to s3
+      await uploadBufferToS3(dwcArchiveZip.toBuffer(), 'application/zip', outputS3Key);
+
+      const connection = getDBConnection(req['keycloak_token']);
+
+      try {
+        await connection.open();
+
+        // Update occurrence submission record to include the transformed output file name and s3 key
+        await updateSurveyOccurrenceSubmissionWithOutputKey(
+          req.body.occurrence_submission_id,
+          outputFileName,
+          outputS3Key,
+          connection
+        );
+
+        await insertSubmissionStatus(
+          req.body.occurrence_submission_id,
+          SUBMISSION_STATUS_TYPE.TEMPLATE_TRANSFORMED,
+          connection
+        );
+
+        next();
+      } catch (error) {
+        await connection.rollback();
+        throw error;
+      } finally {
+        connection.release();
+      }
+    } catch (error) {
+      defaultLog.debug({ label: 'persistTransformationResults', message: 'error', error });
+      throw error;
+    }
+  };
+}
+
+function sendResponse(): RequestHandler {
```

Comment:
can this be defined once somewhere and just exported and used amongst various endpoints?

Reply:
Yes, but I was holding off for now.  I want to move all of our re-useable request handlers into some other folder.  Primarily because you run into circular-dependencies very quickly if all of our endpoints are importing from one-another.

Reply:
Ill try just keeping the version in dwc/validate works, for now.

Reply:
Fixed